chain/network/src/snapshot_hosts/mod.rs

```diff
@@ -193,6 +188,29 @@ impl Inner {
                 }
             }
         }
+    }
+
+    /// Given a state header request produced by the local node,
+    /// selects a host to which the request should be routed.
+    pub fn select_host_for_header(
+        &mut self,
+        sync_hash: &CryptoHash,
+        shard_id: ShardId,
+    ) -> Option<PeerId> {
+        self.maybe_update_sync_hash(sync_hash);
+
+        self.hosts_for_shard.get(&shard_id)?.iter().choose(&mut thread_rng()).cloned()
+    }
+
+    /// Given a state part request produced by the local node,
+    /// selects a host to which the request should be routed.
+    pub fn select_host_for_part(
+        &mut self,
+        sync_hash: &CryptoHash,
+        shard_id: ShardId,
+        part_id: u64,
+    ) -> Option<PeerId> {
+        self.maybe_update_sync_hash(sync_hash);
```

Comment:
sanity check - could this be abused e.g. malicious node sending an arbitrary sync hash to trigger this? 
note - I don't think so since it's still pretty cheap

Reply:
All of the calls to this function are triggered internally; the local node realizes it needs to state sync and triggers some event containing a sync hash. These don't arrive from network peers.